### ISSUE TITLE
Master build fails since October 

### ISSUE THREAD

**Issue by laeubi** [eclipse-platform/eclipse.platform.releng.aggregator#1673]
See https://ci.eclipse.org/platform/job/eclipse.platform.releng.aggregator/job/master/

Last successful was Okt 30 23:04

**Comment by HannesWell** [comment]
Before the last change the error was:

```
08:47:33  [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.5-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.tests.binaries: Version has moved backwards for (org.eclipse.jdt.core.tests.binaries/1.0.250.qualifier). Baseline has 1.0.250.v20210713-1056) with delta: 0.0.0 -> [Help 1]
08:47:33  org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.5-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.tests.binaries: Version has moved backwards for (org.eclipse.jdt.core.tests.binaries/1.0.250.qualifier). Baseline has 1.0.250.v20210713-1056) with delta: 0.0.0
08:47:33      at org.apache.maven.lifecycle.internal
```

So either Tycho fails to resolve the qualifier or the tests bundle needs a version bump.

**Comment by HannesWell** [comment]
Since https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1672 it fails after three or four min with 
```
00:17:22  [INFO] ---------------< org.eclipse.platform:org.eclipse.jface >---------------
00:17:22  [INFO] Building [bundle] JFace 3.33.0-SNAPSHOT                         [27/521]
00:17:22  [INFO]   from eclipse.platform.ui/bundles/org.eclipse.jface/.polyglot.META-INF
00:17:22  [INFO] ---------------------------[ eclipse-plugin ]---------------------------
00:17:22  [INFO] {osgi.os=linux, osgi.ws=gtk, org.eclipse.update.install.features=true, osgi.arch=x86_64}
00:17:22  [ERROR] Cannot resolve project dependencies:
00:17:22  [ERROR]   Software being installed: org.eclipse.jface 3.33.0.qualifier
00:17:22  [ERROR]   Missing requirement for filter properties ~= $0: org.eclipse.swt 3.125.0.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.swt.gtk.linux.x86_64 [3.125.0.qualifier,3.125.0.qualifier]' but it could not be found
00:17:22  [ERROR]   Cannot satisfy dependency: org.eclipse.jface 3.33.0.qualifier depends on: osgi.bundle; org.eclipse.swt [3.111.0,4.0.0)
```

before it failed after ~17min with the cause from my previous comment.

```
08:47:33  [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.5-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.tests.binaries: Version has moved backwards for (org.eclipse.jdt.core.tests.binaries/1.0.250.qualifier). Baseline has 1.0.250.v20210713-1056) with delta: 0.0.0 -> [Help 1]
08:47:33  org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.5-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.tests.binaries: Version has moved backwards for (org.eclipse.jdt.core.tests.binaries/1.0.250.qualifier). Baseline has 1.0.250.v20210713-1056) with delta: 0.0.0
08:47:33      at org.apache.maven.lifecycle.internal
```

So something changed recently, but https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/1672 does not seem really related. And since the I-build failed similar to tonight and @laeubi has written that it is probably a Tycho issue, I assume it's the same here.

**Comment by merks** [comment]
I need to get the TP updated and one cannot do a verification build because it too fails this same way:

https://ci.eclipse.org/platform/job/eclipse.platform.releng.aggregator/job/PR-1693/

```
05:47:06  [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.5-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.tests.binaries: Version has moved backwards for (org.eclipse.jdt.core.tests.binaries/1.0.250.qualifier). Baseline has 1.0.250.v20210713-1056) with delta: 0.0.0 -> [Help 1]
05:47:06  org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.5-SNAPSHOT:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.tests.binaries: Version has moved backwards for (org.eclipse.jdt.core.tests.binaries/1.0.250.qualifier). Baseline has 1.0.250.v20210713-1056) with delta: 0.0.0
```

**Comment by laeubi** [comment]
@akurtakov as the only difference beteen PR (=works) and master (=fails) is the `-Peclipse-sign` I wonder two things:

1. is it useful to sign at all / can it be the reason for issues
2. The jenkins files has a prepare GPG step that seems to be executed always (not only on master) should it probably be skipped for PRs?

**Comment by akurtakov** [comment]
GPG step is(was?) needed for for repo creation which didn't had gpg-signing 3rd party artifacts conditionalized in a profile it's is quite likely still the case. I think we added this signing in order to have this verification build complaining if there is signing issue (even mac signer down) so we catch them earlier. Of course if not working and not "spamming"  - we failed here as it went unnoticed. So it still makes sense to try having signing, if we can't - accept it and skip it.
